src/App.tsx

```diff
@@ -1,223 +1,253 @@
+import { TodoList } from './components/TodoList';
+import './styles/App.scss';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [selectedTodoId, setSelectedTodoId] = useState<number | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isRemoveLoading, setIsRemoveLoading] = useState(false);
+  const [error, setError] = useState('');
+  const [isUpdateLoading, setIsUpdateLoading] = useState(false);
+  const [todosState, setTodosState] = useState('all');
+  const [shouldUpdate, setShouldUpdate] = useState(false);
+  const [isAllToggled, setIsAllToggled] = useState(false);
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+
+  const errorSwitcher = (errorText: string) => {
+    setError(errorText);
+
+    setTimeout(() => setError(''), 3000);
+  };
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    if (user) {
+      setUserId(user.id);
+
+      getTodos(user.id)
+        .then(res => {
+          setTodos(res);
+        })
+        .finally(() => {
+          setShouldUpdate(false);
+        });
+    }
+  }, [user, shouldUpdate, isAllToggled]);
+
+  const addNewTodo = useCallback(((event: FormEvent) => {
+    event.preventDefault();
+
+    if (!todoTitle.trim()) {
+      errorSwitcher('Title can\'t be empty');
+
+      return;
+    }
+
+    setIsLoading(true);
+
+    if (user) {
+      addTodo(todoTitle, userId, false)
+        .then(res => setTodos(prev => [...prev, res]))
+        .catch(() => errorSwitcher('Unable to add a todo'))
+        .finally(() => {
+          setTodoTitle('');
+          setIsAllToggled(false);
+          setIsLoading(false);
+        });
+    }
+  }), [todoTitle, user]);
+
+  const updateTodoById = (todoId: number, data: {}) => {
+    setSelectedTodoId(todoId);
+    setIsUpdateLoading(true);
+    updateTodo(todoId, data)
+      .then(() => {
+        setShouldUpdate(true);
+      })
+      .catch(() => errorSwitcher('Unable to update a todo'))
+      .finally(() => {
+        setIsUpdateLoading(false);
+        setIsAllToggled(false);
+      });
+  };
+
+  const toggleAllCompleted = () => {
+    setIsAllToggled(true);
+
+    if (todos.some(todo => !todo.completed)) {
+      todos.forEach(todo => {
+        if (!todo.completed) {
+          updateTodoById(todo.id, { completed: true });
+          setIsUpdateLoading(true);
+        }
+      });
+    } else {
+      todos.forEach(todo => {
+        updateTodoById(todo.id, { completed: false });
+        setIsUpdateLoading(true);
+      });
     }
-  }, []);
+
+    setShouldUpdate(true);
+  };
+
+  const removeTodo = (todoId: number) => {
```

Comment:
You should wrap all callback with `useCallback` that you pass as props to another component